plugins/inputs/apache/apache.go

```diff
@@ -61,21 +91,53 @@ func (n *Apache) Gather(acc telegraf.Accumulator) error {
 	return outerr
 }
 
-var tr = &http.Transport{
-	ResponseHeaderTimeout: time.Duration(3 * time.Second),
-}
+func (n *Apache) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
+	var err error
+	var resp *http.Response
+	var req *http.Request
+	var tr *http.Transport
+	var client *http.Client
+	var tlsCfg *tls.Config
+
+	if addr.Scheme == "https" {
+		tlsCfg, err = internal.GetTLSConfig(
+			n.SSLCert, n.SSLKey, n.SSLCA, n.InsecureSkipVerify)
+		if err != nil {
+			return err
+		}
+		//fmt.Printf("SSL skip verify = true\n")
+		tr = &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+			TLSClientConfig:       tlsCfg,
+		}
+	} else {
+		//fmt.Printf("no verify configured\n")
+		tr = &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+		}
+	}
 
-var client = &http.Client{
-	Transport: tr,
-	Timeout:   time.Duration(4 * time.Second),
-}
+	client = &http.Client{
+		Transport: tr,
+		Timeout:   n.ResponseTimeout.Duration,
+	}
 
-func (n *Apache) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
-	resp, err := client.Get(addr.String())
-	if err != nil {
-		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	if len(n.Username) != 0 && len(n.Password) != 0 {
+		req, err = http.NewRequest("GET", addr.String(), nil)
+		req.SetBasicAuth(n.Username, n.Password)
+		resp, err = client.Do(req)
+		if err != nil {
+			return fmt.Errorf("error making Basic Auth HTTP request to %s (USER:%s|PASSWORD:%s): %s\n", addr.String(), n.Username, n.Password, err)
+		}
+	} else {
+		resp, err = client.Get(addr.String())
```

Comment:
rather than having two separate ways of issuing the GET (client.Do(req) above & client.Get(addr) here), you should always create the request and use client.Do(req)